table/block_based_table_reader.cc

```diff
@@ -1747,6 +1746,60 @@ Status BlockBasedTable::Prefetch(const Slice* const begin,
   return Status::OK();
 }
 
+Status BlockBasedTable::VerifyChecksum() {
+  Status s;
+  // Check Meta blocks
+  std::unique_ptr<Block> meta;
+  std::unique_ptr<InternalIterator> meta_iter;
+  s = ReadMetaBlock(rep_, &meta, &meta_iter);
+  if (s.ok()) {
+    s = VerifyChecksumInBlocks(meta_iter.get());
+    if (!s.ok()) {
+      return s;
+    }
+  } else {
+    return s;
+  }
+  // Check Data blocks
+  BlockIter iiter_on_stack;
+  InternalIterator* iiter = NewIndexIterator(ReadOptions(), &iiter_on_stack);
+  std::unique_ptr<InternalIterator> iiter_unique_ptr;
+  if (iiter != &iiter_on_stack) {
```

Comment:
It is an ugly hack. After PartitionIndexReader it is no longer guaranteed. If it is normal index, the on-stack object will be used but that on-stack object is not usable for partitioned indexes. So in those cases a new object will be created.

Yi was suggesting to remove the on-stack object altogether but I was hesitant as I was not sure about the performance side-effects.

This is documented in IndexReader::NewIterator but it seems that it should be restated in NewIndexIterator as well.